docs/DeveloperGuide.md

```diff
@@ -1,38 +1,1282 @@
+
+##### Saving Data Sequence Diagram
+![StorageSequenceSleep.png](diagrams/storage/StorageSequenceSleep.png)
+
+* Note: HabitTrackerStorage works in a similar way as SleepTrackerStorage
+
+When `SleepTracker` executes `saveSleepListToFile` method from `SleepTrackerStorage` class, `ArrayList<String>` is 
+created and after successfully converting sleepCycleList to String format, SleepTrackerStorage then calls 
+`saveTasksToFile` from `Storage` class to write data into text file.
+
+### Reflection component
+#### Description
+The Reflection component provides users with random reflections questions to reflect on about their personal lives and goals. 
+This component aims to contribute to the goal of improving user's wellness.
+
+#### Design Considerations
+* ##### User Design Considerations
+  * The sets of 5 questions are randomly generated each time. They are randomised to allow users to reflect on different areas. 
+  * Users can save questions they liked into their favourites list for future review.
+  * Users can also delete questions from their favourites list if the question is no longer relevant to them.
+  * Help menu is available for new users to guide them on how to use the reflection feature.
+  * Error messages with guidance messages will be printed to console if command input by user is invalid.
+
+
+* #### Developer Design Considerations
+  * Modularity: Organized into packages for separation of concerns. Each package contains related classes, such as ReflectionManager and ReflectionList in the reflection package, and ReflectionCommandParser in the parser package.
+  * Abstraction: Use of abstract classes and interfaces for flexibility and extensibility. Abstract classes (ReflectionList) provide a common structure and behavior for subclasses (FavoriteReflectionsList, ReflectionQuestionBank), promoting code reuse and ensuring consistent functionality across different implementations.
+  * Encapsulation: Private access modifiers and encapsulated methods ensure data integrity. Methods like addReflectionQuestion() and removeReflectionQuestion() in ReflectionList encapsulate the manipulation of the reflection list, ensuring data integrity and promoting a clear interface for interacting with the list.
+  * Exception Handling: Graceful error handling with meaningful messages.
+  * Command Design Pattern: Encapsulation of actions as objects for decoupling.
+  * Dependency Injection: Dependencies injected via constructors for loose coupling.
+  * SRP: Classes adhere to the Single Responsibility Principle. For example, the ReflectionManager class is responsible for managing reflection-related operations, while the ReflectionCommandParser class handles parsing and determining reflection commands.
+  * Readability and Maintainability: Descriptive naming, comments, and documentation for clarity.
+
+#### Implementation
+
+##### Class Diagram
+![ReflectionClassDiagram.png](diagrams/reflection/ReflectionClassDiagram.png)
```

Comment:
Might want to simplify or make the diagram's font larger. The class diagram is a little hard to see and is quite convoluted